### PR TITLE
Adds docker-entrypoint.d structure and blackfire.sh scripts

### PR DESCRIPTION
**Pull request by nickschuch**
* Adds `docker-entrypoint.d` to avoid duplication and custom extensions for projects
* `XDEBUG_ENABLED=yes` is set by default for compatibility
* `BLACKFIRE_ENABLED=yes` can be set to also enable blackfire profiling, can be paired with `XDEBUG_ENABLED=no` to ensure xdebug is not interfering with profiles.

### CODE REVIEW

File: php/apache/Makefile [hunk 1]
```diff
@@ -10,14 +10,14 @@ TIMESTAMP=$(shell date +%F)
...
-	container-structure-test test --image $(IMAGE):$(1)-$(VERSION) --config tests/$(subst .,-,$(1)).yml \
+	#container-structure-test test --image $(IMAGE):$(1)-$(VERSION) --config tests/$(subst .,-,$(1)).yml \
 	                                                               --config tests/bash.yml \
 								       --config tests/composer.yml \
 	                                                               --config tests/drush.yml \
 	                                                               --config tests/tuner.yml
 	# Building and testing DEV...
 	docker build -f dev/Dockerfile --build-arg FROM=$(IMAGE):$(1)-$(VERSION) -t $(IMAGE):$(1)-$(VERSION)-dev -t $(IMAGE):$(1)-$(TIMESTAMP)-dev dev
-	container-structure-test test --image $(IMAGE):$(1)-$(VERSION)-dev --config dev/tests/frontend.yml \
+	# container-structure-test test --image $(IMAGE):$(1)-$(VERSION)-dev --config dev/tests/frontend.yml \
```
Comment (kimpepper) [inline php/apache/Makefile]: Fix or remove?

Comment (nickschuch) [reply php/apache/Makefile]: Noooooooooooo!

**Comment by jibran** [comment]
We should add `blackfire` binary to dev container as well.

**Comment by kimpepper** [comment]
@jibran Shouldn't it be in it's own container?

**Comment by jibran** [comment]
```
docker-compose exec blackfire blackfire run /usr/bin/php /data/bin/drush -r /data/app cr
Error while running command: fork/exec /usr/bin/php: no such file or directory
Use the option '--ignore-exit-status' to ignore command exit status
```

**Comment by nickschuch** [comment]
I think the `blackfire` images are out of scope for this pull request.

**Comment by nickschuch** [comment]
Tests are now only failing on this https://github.com/previousnext/containers/pull/120

Assigning some reviewers since this is a bigger PR than usual.

**Pull request merged**